ambry-tools/src/main/java/com.github.ambry.tools/admin/DumpDataHelper.java

```diff
@@ -0,0 +1,630 @@
+        BlobData blobData = MessageFormatRecord.deserializeBlob(streamlog);
+        blobDataOutput = "Blob - size " + blobData.getSize();
+      } else {
+        boolean deleteFlag = MessageFormatRecord.deserializeDeleteRecord(streamlog);
+        isDeleted = true;
+        deleteMsg = "delete change " + deleteFlag;
+      }
+    } else {
+      throw new MessageFormatException("Header version not supported " + version, MessageFormatErrorCodes.IO_Error);
+    }
+    return new BlobRecordInfo(messageheader, blobId, blobProperty, usermetadata, blobDataOutput, deleteMsg, isDeleted,
+        isExpired, timeToLiveInSeconds, totalRecordSize);
+  }
+
+  /**
+   * Dumps replica token file
+   * @param replicaTokenFile the file that needs to be parsed for
+   * @throws Exception
+   */
+  public void dumpReplicaToken(File replicaTokenFile)
+      throws Exception {
+    logOutput("Dumping replica token", LogLevel.INFO);
+    DataInputStream stream = new DataInputStream(new FileInputStream(replicaTokenFile));
+    short version = stream.readShort();
+    switch (version) {
+      case 0:
+        int Crc_Size = 8;
+        StoreKeyFactory storeKeyFactory = Utils.getObj("com.github.ambry.commons.BlobIdFactory", _clusterMap);
+        FindTokenFactory findTokenFactory =
+            Utils.getObj("com.github.ambry.store.StoreFindTokenFactory", storeKeyFactory);
+        while (stream.available() > Crc_Size) {
+          // read partition id
+          PartitionId partitionId = _clusterMap.getPartitionIdFromStream(stream);
+          // read remote node host name
+          String hostname = Utils.readIntString(stream);
+          // read remote replica path
+          String replicaPath = Utils.readIntString(stream);
+          // read remote port
+          int port = stream.readInt();
+          // read total bytes read from local store
+          long totalBytesReadFromLocalStore = stream.readLong();
+          // read replica token
+          FindToken token = findTokenFactory.getFindToken(stream);
+          logOutput(
+              "partitionId " + partitionId + " hostname " + hostname + " replicaPath " + replicaPath + " port " + port
+                  + " totalBytesReadFromLocalStore " + totalBytesReadFromLocalStore + " token " + token, LogLevel.INFO);
+        }
+        logOutput("crc " + stream.readLong(), LogLevel.INFO);
+    }
+  }
+
+  /**
+   * Dumps a single record from the log at a given offset
+   * @param randomAccessFile the {@link RandomAccessFile} referring to log file that needs to be parsed
+   * @param offset the offset at which the record needs to be parsed for
+   * @param blobId the blobId which that is expected to be matched for the record present at
+   *               <code>offset</code>
+   * @param indexValue the {@link IndexValue} that needs to be compared against
+   * @throws IOException
+   */
+  public boolean readFromLog(RandomAccessFile randomAccessFile, long offset, String blobId, IndexValue indexValue,
```

Comment:
`readFromLog` -> `readFromLogAndCompare`. Also may need to change the doc for this method.

The reason I am making this suggestion is from the caller of this method, it says compare, but there is no comparison in the caller body. Only until I came to this method I realize that the comparison is done here.

Also, maybe `verify` is better than `compare`?